packages/library-base/src/commonMain/kotlin/io/realm/notifications/ObjectChange.kt

```diff
@@ -3,29 +3,15 @@ package io.realm.notifications
 import io.realm.RealmObject
 
 sealed interface ObjectChange<O : RealmObject> {
-    enum class State {
-        INITIAL,
-        UPDATED,
-        DELETED
-    }
-
-    /**
-     * Returns the state represented by this change. See [io.realm.notifications.ObjectChange.State]
-     * for a description of the different states a changeset can be in.
-     */
-    val state: State
-
     /**
      * Returns the newest state of object being observed. `null` is returned if the object
      * has been deleted.
      */
     val obj: O?
 }
-interface InitialObject<O : RealmObject> : ObjectChange<O> {
-    override val obj: O
```

Comment:
I added this so people weren't forced to use null-checks everywhere when interacting with `InitialObject`... I think that by removing this, you introduce this problem?

Reply:
Yes, you are right.